### PR TITLE
Functional tests - Order settings enable/disable terms of service

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       |  1.7.7.x
| Description?  | Enable/Disable terms of service
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/02_orderSettings/01_orderSettings/general/06_termsOfService.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18371)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: tests/puppeteer/pages/FO/orderConfirmation.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ module.exports = class OrderConfirmation extends FOBasePage {
     this.orderConfirmationCardSection = '#content-hook_order_confirmation';
     this.orderConfirmationCardTitleH3 = `${this.orderConfirmationCardSection} h3.card-title`;
     this.orderSummaryContent = '#order-summary-content';
+    this.conditionToApproveCheckbox = '#conditions_to_approve\\[terms-and-conditions\\]';
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/orderConfirmation.js]:
```suggestion
    this.conditionToApproveCheckbox = '#conditions_to_approve[terms-and-conditions]';
```

File: tests/puppeteer/pages/FO/orderConfirmation.js [hunk 2]
```diff
@@ -24,4 +25,12 @@ module.exports = class OrderConfirmation extends FOBasePage {
   isFinalSummaryVisible() {
     return this.elementVisible(this.orderSummaryContent, 2000);
   }
+
+  /**
+   * Check if checkbox of condition to approve is visible
+   * @returns {boolean}
+   */
+  isConditionToApproveCheckboxVisible() {
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/FO/orderConfirmation.js]: I think the selector and the function must be in Checkout page not OrderConfirmation.

File: tests/puppeteer/pages/FO/orderConfirmation.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/FO/orderConfirmation.js]: The selector is not valid without '\\'

File: tests/puppeteer/pages/FO/orderConfirmation.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/FO/orderConfirmation.js]: Nice catch :) yes I will fix it

**Review by Progi1984 — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**